### PR TITLE
LPS-158486 Add CanCreateCommentWithValueInternalIdPreviouslyCreated testcase

### PR DESCRIPTION
**Pull request by mateusralv**
Background:
Create automated test CanCreateCommentWithValueInternalIdPreviouslyCreated

Test Plan:
**Given** blog post is created
**And Given** comment with a custom external reference code is created with a POST request
**When** with POST request I create a comment with a custom external reference code with a value of the internal id of the previously created comment
**Then** comment is being created
**And Then** I can see the custom external reference code in the body response

JIRA Link:
https://issues.liferay.com/browse/LPS-158486

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForComments.testcase [hunk 1]
```diff
@@ -44,6 +44,39 @@ definition {
...
+		}
+
+		task ("And Then can see the custom external reference code in the body response") {
+			var externalReferenceCodes = CommentAPI.getFieldValueOfBlogPostingComments(
+				blogPostingId = "${blogPostingId}",
+				fieldName = "externalReferenceCode");
+
+			TestUtils.assertEquals(
+				actual = "${externalReferenceCodes}",
+				expected = "ercComment,ercComment2");
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForComments.testcase:76]: one of the external reference codes should be the internal id of the previous created one

**Review by magjed4289 — CHANGES_REQUESTED** [review]

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForComments.testcase [hunk 2]
```diff
@@ -44,6 +44,51 @@ definition {
...
+		}
+
+		task ("When with POST request I create a comment with a custom external reference code with a value of the internal id of the previously created comment") {
+			var idComment = JSONUtil.getWithJSONPath("${response}", "$.id");
+
+			var response2 = CommentAPI.createCommentInBlogPosting(
+				blogPostingId = "${blogPostingId}",
+				commentText = "CommentTwo",
+				externalReferenceCode = "${idComment}",
+				idComment = "1");
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForComments.testcase:68]: why do we put an idComment into the createComment endpoint? 🤔

Comment (mateusralv) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForComments.testcase:68]: Ops. I confused the IDs requested in the test, but I will remove and update the PR. 😅

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessdeliveryapi/CommentAPI.macro [hunk 3]
```diff
@@ -68,6 +68,10 @@ definition {
 			var parentCommentId = "0";
 		}
 
+		if (!(isSet(idComment))) {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessdeliveryapi/CommentAPI.macro:71]: this can be removed as well then, right?

**Review by magjed4289 — APPROVED** [review]
Please remove the idComment appearance from the macro layer, as I understand correctly it is no longer needed 😅 . after that, you can forward it! 👌🏻 Thank you!

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by mateusralv** [comment]
ci:forward

**Pull request closed without merging**